lib/cucumber/listener/pretty_formatter.js

```diff
@@ -48,7 +48,12 @@ var PrettyFormatter = function(options) {
     source = tagSource + self._pad(source, currentMaxStepLength + 3);
 
     var uri = color.format('comment', "# " + scenario.getUri().replace(process.cwd(),'').slice(1) + ":" + scenario.getLine());
-    source += uri + "\n";
+
+    if (options.showSource) {
+        source += uri + "\n";
+    } else {
+        source += "\n";
+    }
```

Comment:
can dry this up with
```js
if (options.showSource) {
  source += uri;
}
source += "\n";
```
Also would be good to put the declaration of the uri variable in the if statement